src/nvim/ex_docmd.c

```diff
@@ -8654,7 +8654,10 @@ eval_vars (
       break;
 
     case SPEC_AFILE:  // file name for autocommand
-      if (autocmd_fname != NULL && !path_is_absolute(autocmd_fname)) {
+      // For CmdlineEnter and related events, <afile> is not a path! #9348
+      if (autocmd_fname != NULL
+          && !path_is_absolute(autocmd_fname)
+          && !strequal("/", (char *)autocmd_fname)) {
```

Comment:
the comment should be attached to this line, it is not relevant to the other conditions